src/Contacts/src/GlobalCoPEvaluator.cpp

```diff
@@ -0,0 +1,178 @@
+{
+    // save the contacts
+    m_contacts = input;
+    return true;
+}
+
+bool GlobalCoPEvaluator::isOutputValid() const
+{
+    return m_isOutputValid;
+}
+
+const Eigen::Vector3d& GlobalCoPEvaluator::getOutput() const
+{
+    return m_cop;
+}
+
+bool GlobalCoPEvaluator::advance()
+{
+    constexpr auto logPrefix = "[GlobalCoPEvaluator::advance]";
+    m_isOutputValid = false;
+
+    if (!m_isInitialized)
+    {
+        log()->error("{} The object is not initialized. Please call "
+                     "GlobalCoPEvaluator::initialize().",
+                     logPrefix);
+        return false;
+    }
+
+    Eigen::Vector3d globalCoP;
+    Eigen::Vector3d localCoP;
+    double totalForce = 0;
+    int numberOfActiveSupports = 0;
+    for (const auto& contact : m_contacts)
+    {
+        // check if the z component of the force is greater than a threshold
+        if (contact.wrench.force()[2] < m_minimumNormalForce)
+        {
+            continue;
+        }
+
+        // the local CoP is defined since fz is greater than zero
+        localCoP = contact.wrench.getLocalCoP();
+
+        // check if the CoP is outside the admissible limits
+        if (std::abs(localCoP[0]) > m_CoPAdmissibleLimits[0]
+            || std::abs(localCoP[1]) > m_CoPAdmissibleLimits[1])
+        {
+            continue;
+        }
+
+        globalCoP += contact.wrench.force()[2] * contact.pose.act(localCoP);
+        totalForce += contact.wrench.force()[2];
+
+        numberOfActiveSupports++;
+    }
+
+    if (numberOfActiveSupports == 0)
+    {
+        log()->error("{} Zero contacts are active the CoP is not valid.", logPrefix);
+        return false;
+    }
+
+    // compute the global CoP.
+    globalCoP /= totalForce;
+
+    // at least two contacts are active and the counter is active
+    if (numberOfActiveSupports > 1 && m_constantCoPMaxCounter > 0)
+    {
+        const double CoPDifference = (globalCoP - m_cop).norm();
+
+        // check if the CoP is constant
+        if (CoPDifference < m_constantCoPTolerance)
+        {
+            // CoP considered constant so increase the counter
+            m_constantCoPCounter++;
+        } else
+        {
+            // CoP is not constant so reset the counter
+            m_constantCoPCounter = 0;
+        }
+
+        if (m_constantCoPCounter >= m_constantCoPMaxCounter / 2)
+        {
+            log()->warn("{} The CoP was constant (in a {}m radius) for {} times.",
+                        logPrefix,
+                        m_constantCoPTolerance,
+                        m_constantCoPCounter);
+        }
+
+        if (m_constantCoPCounter >= m_constantCoPMaxCounter)
+        {
+            log()->error("{} The CoP was constant (in a {} m radius) for {} times.",
+                         logPrefix,
+                         m_constantCoPTolerance,
+                         m_constantCoPCounter);
+
+            return false;
+        }
```

Comment:
Maybe, we should consider the case in which ``m_constantCoPMaxCounter`` is negative. In the walking controller, if the max counter was negative, we simply skipped the check.

Reply:
Hi @S-Dafarra it's the same here 
```cpp
    if (numberOfActiveSupports > 1 && m_constantCoPMaxCounter > 0)
    {

```

Reply:
Ah sorry, I did not notice it. Is it worth mentioning this case in the docs?

Reply:
I would say yes

Reply:
Done here: https://github.com/ami-iit/bipedal-locomotion-framework/pull/745/commits/95727d6dd7e4deebbefcfd0e9fe78de93f5522ce